misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParserErrorHandler.java

```diff
@@ -332,7 +362,11 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
             case TOML_TABLE_ARRAY:
                 return ParserRuleContext.ARRAY_TABLE_FIRST_START;
             case COMMA:
-                return ParserRuleContext.ARRAY_VALUE_START;
+                if (parentCtx.equals(ParserRuleContext.INLINE_TABLE_LIST)) {
+                    return ParserRuleContext.INLINE_TABLE_ENTRY_START; //TODO Check
+                } else {
+                    return ParserRuleContext.ARRAY_VALUE_START;
+                }
```

Comment:
`INLINE_TABLE_ENTRY_START` and `ARRAY_VALUE_START` contexts  seem to be redundant. 
```suggestion
                return ParserRuleContext.VALUE;
```